File: solutions/csharp/ReversePolishNotation.cs
public class Solution
{
    public int EvalRPN(string[] tokens)
    {
        Stack<int> stack = new Stack<int>();

        foreach (string token in tokens)
        {
            if (isOperator(token))
            {
                int right = stack.Pop();
                int left = stack.Pop();

                int calculated = Calculate(left, right, token);
                stack.Push(calculated);
            }
            else
            {
                stack.Push(int.Parse(token));
            }
        }

        return stack.Peek();

    }

    public bool isOperator(string str)
    {
        return (
                str == "+" || str == "-" || str == "*" || str == "/"
               );
    }

    public Calculate(int a, int b, string op)
    {
        switch (op)
        {
            case "+":
                return a + b;
            case "-":
                return a - b;
            case "*":
                return a * b;
            case "/":
                return a / b;
            default:
                return 0;
        }
    }
}

File: solutions/java/reversePolishNotation.java
public int evalRPN(String[] tokens) {
	Deque<Integer> stack = new ArrayDeque<Integer>();

	for (String token : tokens) {
		if (isOperator(token)) {
			int tokA = stack.pop();
			int tokB = stack.pop();

			stack.push(evaluate(tokB, tokA, token));
		} else {
			stack.push(Integer.parseInt(token));
		}
	}
	return stack.peek();
}

public Boolean isOperator(String c) {
	return c.equals("+") || c.equals("-") || c.equals("*") || c.equals("/");
}

public int evaluate(int a, int b, String op) {
	switch (op) {
		case "+":
			return a + b;
		case "-":
			return a - b;
		case "*":
			return a * b;
		case "/":
			return a / b;
		default:
			System.out.println("Not a valid operator!");
	}
	return 0;
}
